ui/src/app/settings/selectors/general/general.js

```diff
@@ -0,0 +1,46 @@
+const general = {};
+
+/**
+ * Returns object with osinfo data
+ * @param {Object} state - the redux state
+ * @returns {Object} The osinfo data
+ */
+general.osinfo = state => state.general.items;
+
+/**
+ * Returns true if general is loading
+ * @param {Object} state - the redux state
+ * @returns {Boolean} - general is loading
+ */
+general.loading = state => state.general.loading;
+
+/**
+ * Returns true if general is loaded
+ * @param {Object} state - the redux state
+ * @returns {Boolean} - general is loading
+ */
+general.loaded = state => state.general.loaded;
+
+/**
+ * Returns kernels data
+ * @param {Object} state - the redux state
+ * @returns {Array} the available kernel options
+ */
+general.getKernalOptionsByKey = (state, key) => {
+  let kernelOptions = [];
+
+  if (state.general.items.kernels) {
+    kernelOptions = state.general.items.kernels.ubuntu[key];
```

Comment:
This will break the UI if state.general.items.kernels.ubuntu doesn't exist.

I think this needs to be rewritten because really it requires knowledge of two things: the release *and* the OS. Otherwise it's only returning Ubuntu kernels.

Reply:
The field that this populates only requires ubuntu kernels so won't need knowledge of the os

Reply:
Ahh, right. In that case I would name the selector `getUbuntuKernelOptions` or something, and I would still add a check for `state.general.kernels.ubuntu`.

Reply:
Done